hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/HazelcastOperatorTableVisitor.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate;
+
+import com.hazelcast.sql.impl.calcite.validate.operators.HazelcastSqlCase;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.fun.SqlCase;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.util.SqlBasicVisitor;
+import org.apache.calcite.sql.validate.SqlNameMatcher;
+import org.apache.calcite.sql.validate.SqlNameMatchers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Rewrites operators in SqlNode tree from Calcite ones to Hazelcast ones.
+ *
+ * @see SqlStdOperatorTable
+ * @see HazelcastSqlOperatorTable
+ */
+public final class HazelcastOperatorTableVisitor extends SqlBasicVisitor<Void> {
```

Comment:
Can we document why we rewrite `CASE` and do the overload lookup? What would happen if we do not have this class?

Reply:
fixed in 7f2b7d16e9d13cd0e6f7f8586a512f9252fa4fbf